tests/unit/customizations/cloudformation/test_deploy.py

```diff
@@ -60,6 +60,7 @@ def setUp(self):
                                     fail_on_empty_changeset=True,
                                     s3_bucket=None,
                                     s3_prefix="some prefix",
+                                    s3_endpoint_url="http://localhost",
```

Comment:
I would like to see tests explicit tests for this behavior.

At the least there needs to be :
1) Test where argument is not supplied showing that the old endpoint behavior used.
2) If we do supply an override argument, that is correctly passed through and used in the request.